### ISSUE TITLE
Error in project build after updating to 9.0.4 from 9.0.2 (regression??)

### ISSUE THREAD

**Issue by AcidSlide** [angular/angular#35770]
# 🐞 bug report
Prior to updating 9.0.4, running `ng build` and `ng serve` doesn't encounter any compilation error, but after updating I'm getting `Javascript heap out of memory`.

The weird part is, clearing/deleting `node_modules` and re-installing via `npm install`, the compile error goes away.

NOTE: Actually I posted already that I was getting this error yesterday, but somehow the issue got deleted by somebody.

### Affected Package
@angular/core 9.0.4

### Is this a regression?
I'm not sure if this should be treated as regression or an actual bug, since build failure goes away after deleting `node_modules` and re-installing via `npm install`.

### Description
Prior to updating to @angular/core 9.0.4, no issues upon running `ng build` and/or `ng serve`

Below is prior to updating to 9.0.4
![image](https://user-images.githubusercontent.com/6833047/75599799-58ac8a00-5ae3-11ea-9fb2-e35c2277c371.png)
(Angular Core version 9.0.2)
![image](https://user-images.githubusercontent.com/6833047/75599831-aa551480-5ae3-11ea-8f4f-1d80b5e2c7d3.png)

After updating to 9.0.4 this happens running `ng build`
![image](https://user-images.githubusercontent.com/6833047/75599878-01f38000-5ae4-11ea-993c-00c44cd6b829.png)

BUT..... (a big BUT!!)

I tried the following, and suddenly `ng build` and/or `ng server` works 
I tried deleting `node_module` directory and re-installed the node packages `npm install` and now the build error is gone!?

## 🔬 Minimal Reproduction
Currently, I don't have a minimal reproduction repo available, the project is already big as of now so creating a minimal reproduction is out of the picture due lack of time to do it.

## 🔥 Exception or Error
![image](https://user-images.githubusercontent.com/6833047/75599878-01f38000-5ae4-11ea-993c-00c44cd6b829.png)

## 🌍  Your Environment
Prior version
![image](https://user-images.githubusercontent.com/6833047/75599799-58ac8a00-5ae3-11ea-9fb2-e35c2277c371.png)

**Angular Version:**
<pre><code>
     _                      _                 ____ _     ___
    / \   _ __   __ _ _   _| | __ _ _ __     / ___| |   |_ _|
   / △ \ | '_ \ / _` | | | | |/ _` | '__|   | |   | |    | |
  / ___ \| | | | (_| | |_| | | (_| | |      | |___| |___ | |
 /_/   \_\_| |_|\__, |\__,_|_|\__,_|_|       \____|_____|___|
                |___/
    

Angular CLI: 9.0.4
Node: 10.19.0
OS: darwin x64

Angular: 9.0.4
... animations, cli, common, compiler, compiler-cli, core, forms
... language-service, localize, platform-browser
... platform-browser-dynamic, router
Ivy Workspace: Yes

Package                           Version
-----------------------------------------------------------
@angular-devkit/architect         0.900.4
@angular-devkit/build-angular     0.900.4
@angular-devkit/build-optimizer   0.900.4
@angular-devkit/build-webpack     0.900.4
@angular-devkit/core              9.0.4
@angular-devkit/schematics        9.0.4
@ngtools/webpack                  9.0.4
@schematics/angular               9.0.4
@schematics/update                0.900.4
rxjs                              6.5.4
typescript                        3.8.2
webpack                           4.41.2
</code></pre>

**Anything else relevant?**
I'm not sure if this was related to the build issues with core version 9.0.3.

**Comment by ericmartinezr** [comment]
Try upgrading your node version to 12. Also check cli's repo for this error, is a very well known message.

**Comment by AcidSlide** [comment]
> Try upgrading your node version to 12. Also check cli's repo for this error, is a very well known message.

I can't, our production and test bed is using version 10, and this is just a small portion of the whole project. There are several projects under this and all using version 10.

**Comment by AcidSlide** [comment]
Actually, the more pressing question is, why did clearing the `node_modules` and re-installing them worked??

**Comment by igorosabel** [comment]
Same here. I have some projects, all of them with Angular 9.0.2 that build correctly. After doing “ng update @angular/core” they get bumped to 9.0.4 and then start failing.

Running “ng build —prod” results in “Error in stack call size”.

But as @AcidSlide pointed, after the “ng update” I deleted the “node_modules” folder, run “npm install” and inmedeately after “ng build” and worked flawlessly.

I have Node 12 and I have reproduced this error on Windows, Mac and Linux machines. If you want a repo to reproduce this error try one of mine ( https://github.com/igorosabel/Void )

Thanks!
